test/src/unittests/highlevel/test_crosssimilaritymatrix.py

```diff
@@ -18,7 +18,6 @@
 # version 3 along with this program. If not, see http://www.gnu.org/licenses/
 
 from sklearn.metrics.pairwise import euclidean_distances
```

Comment:
This is the only place where sklearn is used in Essentia. We should at least replace it for [scipy](with https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.distance_matrix.html), or even re-implement from scratch. 

Even better, we can also use a pre-computed answer. In the case we use simple matrices in the multiplication that's easy to hardcode in the code. Otherwise, load from a file. Are current query and reference features in the tests the best to detect possible issues? One in the matrix of ones, another one is matrix of zeros.

Reply:
Replaced query and reference hpcp matrix values with the ones in `test_chromacrosssimilaritymatrix. Added pre-computed result for the cross similarity matrix and thus removed the extra overhead of computing it in the test as you suggested (also the sklearn dependency).